client/src/pages/signUp/SignUp.tsx

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable max-len */
+import React, { useState } from "react";
+import { Form, Radio, Button, message } from "antd";
+import "antd/dist/antd.min.css";
+import "./SignUp.css";
+import axios from "axios";
+import { StudentSignUp, ParentSignUp, TeacherSignUp } from "../../components";
+
+interface signUpDataInterface {
+  name: string;
+  email: string;
+  mobile: string;
+  password: string;
+  confPassword: string;
+  location: string;
+  role: string;
+  children: string[];
+}
+
+const init = {
+  name: "",
+  email: "",
+  mobile: "",
+  password: "",
+  confPassword: "",
+  location: "",
+  role: "",
+  children: [""],
+};
+
+const SignUpPage: React.FC = () => {
+  const [role, setRole] = useState<string>("teacher");
+  const [signUpData, setSignUpData] = useState<signUpDataInterface>(init);
+
+  const handleRoleValue: any = (e: any) => {
+    setRole(e.target.value);
+    init.role = role;
+  };
+
+  const addData: any = async (data: signUpDataInterface) => {
+    try {
+      await axios.post("/api/v1/signup", data);
+    } catch (err: any) {
+      message.error(err);
+    }
```

Comment:
We send a message in the back (msg), with axios, you can access it through `err.response.data.msg`